consul/prepared_query_endpoint.go

```diff
@@ -562,6 +572,18 @@ func tagFilter(tags []string, nodes structs.CheckServiceNodes) structs.CheckServ
 	return nodes[:n]
 }
 
+// nodeMetaFilter returns a list of the nodes who satisfy the given metadata filters. Nodes
+// must have ALL the given tags.
+func nodeMetaFilter(filters map[string]string, nodes structs.CheckServiceNodes) structs.CheckServiceNodes {
```

Comment:
It seems like a tradeoff; if we do it this way then we're doing less allocations/copying for the case where most of the nodes get filtered out, and if we do it the filterHealthChecks way, we have less allocations/copying for the case where most nodes aren't filtered out. I don't know if one case is more common because it depends on the user's setup.